acceptance: TestDockerCLI_test_demo_node_cmds failed [node reports draining=true after restart is completed]

acceptance.TestDockerCLI_test_demo_node_cmds [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/14783942?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/14783942?buildTab=artifacts#/) on release-24.1 @ [64f3f0bfab805c0bad14c2dc26b47693b0729140](https://github.com/cockroachdb/cockroach/commits/64f3f0bfab805c0bad14c2dc26b47693b0729140):


```
=== RUN   TestDockerCLI_test_demo_node_cmds
    test_log_scope.go:170: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/_tmp/b38fdc3e3d87d99024418e1ae9af5f4c/logTestDockerCLI_test_demo_node_cmds1946639592
    test_log_scope.go:81: use -show-logs to present logs inline
    generated_cli_test.go:145: -- test log scope end --
test logs left over in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/_tmp/b38fdc3e3d87d99024418e1ae9af5f4c/logTestDockerCLI_test_demo_node_cmds1946639592
--- FAIL: TestDockerCLI_test_demo_node_cmds (69.10s)
=== RUN   TestDockerCLI_test_demo_node_cmds/test_demo_node_cmds.tcl
    util_docker.go:259: non-zero exit code: 1
    --- FAIL: TestDockerCLI_test_demo_node_cmds/test_demo_node_cmds.tcl (59.97s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119508 acceptance: TestDockerCLI_test_demo_node_cmds failed [restart caused &#34;avoid up-replicating to fragile quorum&#34;; store throttled] [A-testing C-bug C-test-failure O-robot P-2 T-kv branch-master]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDockerCLI_test_demo_node_cmds.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-37720

--- a/pkg/cli/interactive_tests/test_demo_node_cmds.tcl
+++ b/pkg/cli/interactive_tests/test_demo_node_cmds.tcl
@@ -59,10 +59,28 @@ send "SELECT count(*) FROM system.users;\r"
 eexpect "1"
 eexpect "defaultdb>"
 
-# Now restart the node.
+# Now restart the node, and wait for liveness to update.
+# See: https://github.com/cockroachdb/cockroach/issues/122134#issuecomment-2050667479.
 send "\\demo restart 3\r"
 eexpect "node 3 has been restarted"
 eexpect "defaultdb>"
+
+set timeout 1
+set stmt "select node_id, draining, membership from crdb_internal.kv_node_liveness ORDER BY node_id;\r"
+send $stmt
+expect {
+    "3 |    f     | active" {
+        puts "\rfound n3 active and no longer draining"
+    }
+    timeout {
+        puts "\rdid not see n3 active and no longer draining - retrying"
+        sleep 2
+        send $stmt
+        exp_continue
+    }
+}
+# Reset timeout back to 45 to match common.tcl.
+set timeout 45
 
 send "select node_id, draining, membership from crdb_internal.kv_node_liveness ORDER BY node_id;\r"
 eexpect "1 |    f     | active"

